### PR TITLE
Revert "Only return active protocol components from z_gettreestate."

### PR DESCRIPTION
**Pull request by nuttycom**
This reverts commit c44e58ea519d3c6fefb59117d9ce610afca22d0f and adds an alternate implementation that only affects Orchard, so as not to disrupt existing API users.

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1337,7 +1337,8 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
     }
 
     // orchard
-    {
+    auto nu5_activation = Params().GetConsensus().GetActivationHeight(Consensus::UPGRADE_NU5);
```
Comment (daira) [inline src/rpc/blockchain.cpp:1340]: `nu5_activation_height`

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -1316,44 +1316,43 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
...
+            s << OrchardMerkleFrontierLegacySer(tree);
+            orchard_commitments.pushKV("finalState", HexStr(s.begin(), s.end()));
+        } else {
+            // Set skipHash to the most recent block that has a finalState.
+            const CBlockIndex* pindex_skip = pindex->pprev;
+            if (pindex_skip && pindex_skip->nHeight >= nu5_activation_height.value()) {
+                while (pindex_skip && pindex_skip->nHeight > nu5_activation_height.value() && !pcoinsTip->GetOrchardAnchorAt(pindex_skip->hashFinalOrchardRoot, tree)) {
                     pindex_skip = pindex_skip->pprev;
                 }
                 if (pindex_skip) {
```
Comment (str4d) [inline src/rpc/blockchain.cpp:1358]:
```suggestion
            auto orchardActive = [&](pindexCur) {
                return pindex_skip && pindex_skip->nHeight >= nu5_activation_height.value();
            };
            while (orchardActive(pindex_skip) && !pcoinsTip->GetOrchardAnchorAt(pindex_skip->hashFinalOrchardRoot, tree)) {
                pindex_skip = pindex_skip->pprev;
            }
            if (orchardActive(pindex_skip)) {
```

**Review by str4d — APPROVED** [review]
utACK

**Review by daira — COMMENTED** [review]
Post-hoc utACK

**Pull request merged**